Modix.WebServer/ModixWebServer.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Antiforgery;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Modix.Data.Models;
+using Modix.Services.CodePaste;
+using Modix.Services.GuildInfo;
+using System;
+using System.IO;
+
+namespace Modix.WebServer
+{
+    public class ModixWebServer
+    {
+        private static IServiceCollection _additionalServices;
+        private static ModixConfig _modixConfig;
+
+        // Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            foreach (var service in _additionalServices)
+            {
+                services.Add(service);
+            }
+
+            services
+            .AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
+            .AddCookie(options =>
+            {
+                options.LoginPath = "/api/unauthorized";
+                //options.LogoutPath = "/logout";
+                options.ExpireTimeSpan = new TimeSpan(7, 0, 0, 0);
+                
+            })
+            .AddDiscord(options =>
+            {
+                options.ClaimActions.MapJsonKey(claimType: "avatarHash", jsonKey: "avatar");
+
+                options.ClientId = _modixConfig.DiscordClientId;
+                options.ClientSecret = _modixConfig.DiscordClientSecret;
+                options.Scope.Add("identify");
+            });
+
+            services.AddAntiforgery(options => options.HeaderName = "X-XSRF-TOKEN");
```

Comment:
I'm actually not seeing any kind of authorization WRT the user logging into the website. Can anyone on the server login, or should role checks be performed? Or did I just miss it?